### PR TITLE
Streamline local dev process with hot reloading, Dockerfile.dev updates, and resolved fixtures

### PR DESCRIPTION
**Pull request by aaronkanzer**
Relates to:

• https://github.com/lincbrain/linc-archive/issues/27

Builds upon:

Cc @kabilar

• https://github.com/lincbrain/linc-archive/pull/26

Cc @yarikoptic @waxlamp 

• https://github.com/dandi/dandi-archive/pull/1722

**Purpose**: 

For the LINC project -- @kabilar and I have been trying to improve the development process. This PR introduces a simple bash script for local setup, hot-reloading on the frontend, and improved fixtures instantiation. 

For the fixtures improvements:

• A minor modification is made to ensure that a `User` object is created if not found -- this sometimes passed; however, sometimes it did not due to the presence of past data in Postgres being present (especially when switching back-and-forth from `linc-archive` to `dandi-archive`) This addition is fairly benign in my opinion

• Extending from https://github.com/dandi/dandi-archive/pull/1722, it was observed that validation failed initially upon the creation of an `AssetBlob`, as with the SimpleUploadedFile as a non-zarr asset, validation fails without a sha_256 initially provided. See the error message below that results from prior these changes

```
[20:14:42] INFO     Found AssetBlob c000611b-2d6c-4931-b0ec-44855ec1d379                                                                                        __init__.py:21
           INFO     Validating asset metadata for asset 1                                                                                                       __init__.py:40
           INFO     Error while validating asset 1 [{'loc': ('digest',), 'msg': 'A non-zarr asset must have a sha2_256.', 'type': 'value_error'}]               __init__.py:57
           INFO     Validating dandiset metadata for version 1                                                                                                 __init__.py:125
           INFO     Successfully validated version 1    
```

### CODE REVIEW

**Comment by aaronkanzer** [comment]
Cc @mvandenburgh -- just tagging you here to get your thoughts/opinions -- this setup is jut something we have found convenience with for the `linc-archive` fork, but if you have future plans/better ideas, let us know. I noticed all of your frontend improvements regarding `vite` builds in the end of 2023 😃

**Comment by yarikoptic** [comment]
I know that it was not reviewed yet but conflicts already came up so might be good to rebase @aaronkanzer

File: admin_dev_startup.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# This script can be used to launch both the frontend and backend with ease locally
+# Happy developing!
+
+if [ $# -lt 2 ]; then
+  echo "Usage: $0 <image_name> <your.email@email.com>"
+  exit 1
+fi
+
```
Comment (yarikoptic) [inline admin_dev_startup.sh:10]:
```suggestion

# Fail as early as any error or any undefined variable
set -eu

```

**Comment by aaronkanzer** [comment]
@yarikoptic thanks for the review here -- I'm navigating over in the Terraform land, so I might not get back to this for a few days -- many thanks in the meantime though

**Comment by yarikoptic** [comment]
since there seems to be no other shell script  in this repo I would not introduce shellcheck workflow here, but here is the script with which I would have done it: https://github.com/yarikoptic/improveit/blob/main/shellcheckit ;-)

**Comment by yarikoptic** [comment]
@mvandenburgh thanks for the review!  I think it is still worth reincarnating this PR and a developer helper, WDYT @aaronkanzer , would you be able to resolve conflicts and reflect on @mvandenburgh comments?

**Comment by aaronkanzer** [comment]
Just re-visiting now -- sorry for the delay, thank you @mvandenburgh @waxlamp and @yarikoptic  for the review.

I think I'm going to close this PR and not move forward to move (I'll just resort to using this as a local utility). I'd rather not make it a configurable script that is merged (seems like an increase in cognitive overhead if this is not unanimously wanted by other folks)

**Pull request closed without merging**